src/components/HomeScreen.js

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import {
+  TextInput,
+  TouchableOpacity,
+  Button,
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { connect } from 'react-redux';
+import {verifyUser} from '../store/UsersActions';
+
+class HomeScreen extends Component {
+  constructor(){
+    super();
+    this.state = {
+      email: "",
+      password: "",
+    }
+    this.onLogin = this.onLogin.bind(this);
+  }
+
+  onLogin (navigation){
+    const expression = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+    if(this.state.password.length > 8 && expression.test(this.state.email.toLowerCase())){
+      this.props.onLogin(this.state.email);
+      navigation.navigate('Details');
+    }
+    else{
+      alert("Invalid email or password");
+    }
+  } 
+
+
+  render(){
+    const { navigation } = this.props;
+    
+
+    return (
+      <View style={styles.container}>
+        <Text style={{ fontSize: 30 }}>This is the home screen!</Text>
+        <Text>You have { this.props.users.available.length } users.</Text>
```

Comment:
A mesma coisa aqui, podes fazer destruct do `users` do props
```suggestion
        <Text>You have {users.available.length} users.</Text>
```